### PR TITLE
allow modify and translate at the same time

### PR DESCRIPTION
**Pull request by waxenegger**
Geometric modification of shapes used to be mutually exclusive with translation. 

Now both can be done without switching modes. 

TEST: use https://cowfish.openmicroscopy.org/webtrial/, select shapes, then modify and translate them.

### CODE REVIEW

**Comment by will-moore** [comment]
This is working nicely now - much improved workflow.
Good to merge.

File: ol3-viewer/src/ome/ol3/Viewer.js [hunk 1]
```diff
@@ -697,9 +697,11 @@ ome.ol3.Viewer.prototype.addRegions = function(options) {
 	// add a vector layer with the regions
 	if (this.regions_) {
 		this.viewer_.addLayer(new ol.layer.Vector({source : this.regions_}));
-		// enable roi selection by default
+		// enable roi selection by default,
```
Comment (jburel) [inline ol3-viewer/src/ome/ol3/Viewer.js]: minor point: could you fix the indentation?
it is a mix of space and tab in this file

File: src/regions/regions-edit.js [hunk 2]
```diff
@@ -5,8 +5,8 @@ import {Utils} from '../utils/regions';
 import {Converters} from '../utils/converters';
 import {REGIONS_MODE} from '../utils/constants';
 import {
-    REGIONS_CHANGE_MODES, REGIONS_SET_PROPERTY,
-    REGIONS_GENERATE_SHAPES,REGIONS_MODIFY_SHAPES, REGIONS_COPY_SHAPES
+    REGIONS_SET_PROPERTY, REGIONS_GENERATE_SHAPES,
```
Comment (jburel) [inline src/regions/regions-edit.js]: maybe alphabetical order to make it easier to read

File: ol3-viewer/src/ome/ol3/Viewer.js (hunk 1 above)
Comment (waxenegger) [inline ol3-viewer/src/ome/ol3/Viewer.js]: sure, as a matter of fact -and I have started this already- if I find myself making changes to a file where there are tabs I replace them from now on. this is only relevant for files that reside in ol3-viewer. it's however somewhat tedious work (like with big files) unless I just leave it at the automatic conversion...

**Comment by waxenegger** [comment]
ok, had to redo all commits and force push because of https://github.com/ome/omero-iviewer/pull/23 merge into master. also added changes suggested by @jburel.

**Comment by will-moore** [comment]
@waxenegger I usually find that merging of 'master' into your branch and fixing of conflicts is usually nicer than rebasing because you don't have to force push (preserves history better).

**Comment by waxenegger** [comment]
well I cherry pick the commits so apart from the ones where I need to merge the changes history will be ok but might just merge master next time...

**Comment by jburel** [comment]
This might a side effect of this work
If I draw 2 rectangles on the same plane
Then click "Select All" in the Edit button
I have a rectangle with white border that does not match the 2 rectangles I have initially on display

**Comment by waxenegger** [comment]
Hmm, I actually have an issue with the "Select All" as well on cowfish but I don't see a white border ....
I'm going to look into it.

**Comment by jburel** [comment]
you are correct 
I was in the drawing mode.
We will need to look in another PR when the ``select all`` is enabled or not

**Comment by jburel** [comment]
The overlapping of shapes will also need to be investigated in another PR. 
I will add it to the card
Otherwise the PR tackles the problem "modify & translate"
Merging

**Pull request merged**